source/MaterialXGenOgsXml/GlslFragmentGenerator.cpp

```diff
@@ -0,0 +1,192 @@
+//
+// TM & (c) 2017 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXGenOgsXml/GlslFragmentGenerator.h>
+
+#include <MaterialXGenShader/Shader.h>
+
+namespace MaterialX
+{
+
+const string GlslFragmentGenerator::TARGET = "ogsxml";
+
+GlslFragmentGenerator::GlslFragmentGenerator() :
+    GlslShaderGenerator()
+{
+}
+
+ShaderGeneratorPtr GlslFragmentGenerator::create()
+{
+    return std::make_shared<GlslFragmentGenerator>();
+}
+
+ShaderPtr GlslFragmentGenerator::generate(const string& name, ElementPtr element, GenContext& context) const
+{
+    ShaderPtr shader = createShader(name, element, context);
+
+    ShaderStage& stage = shader->getStage(Stage::PIXEL);
+
+    // Rename the pixel stage input struct to match OGS.
+    VariableBlock& vertexData = stage.getInputBlock(HW::VERTEX_DATA);
+    vertexData.setInstance("PIX_IN");
```

Comment:
May be better to make it a settable, as I'm not sure if it's always this name or not yet.